Add license info

TheBlueMatt opened the pull request.

valentinewallace commented on c-bindings-gen/src/main.rs:
```diff
@@ -1342,6 +1351,14 @@ fn convert_file<'a, 'b>(libast: &'a FullLibraryAST, crate_types: &mut CrateTypes
+		writeln!(out, "// history and in the source files from which this was generated.").unwrap();
+		writeln!(out, "//").unwrap();
+		writeln!(out, "// This file is licensed under the license available in the LICENSE or LICENSE.md").unwrap();
+		writeln!(out, "// file in the root of this repositry or, if no such file exists, the same license").unwrap();
+		writeln!(out, "// as that which applies to the original source files.").unwrap();
```
nit: repository

Were these sentences pre-ordained by some lawyer somewhere? Just wondering because I'm a bit confused whether "the original souce files" refers to Rust-Lightning or `lightning-c-bindings`, but if it's preapproved legalese that's fine..

TheBlueMatt replied:
Nope, you can tell I wrote it because there's typos in it :).

valentinewallace reviewed (commented):
CI failing for some reason?

Also, wonder if I could get the perms to be able to self-request reviews :grimacing: nbd though

TheBlueMatt commented:
Updated wording slightly and updated the git hash to point to latest upstream (which is why the CI was sad). I would give you access to this repo, but only @moneyball or @arik-so appear to be able to add new org members. We should probably just have a team that is "everyone who has access to stuff" and add people to that.

valentinewallace commented on LICENSE.md:
```diff
@@ -0,0 +1,14 @@
+This software is licensed under [Apache 2.0](LICENSE-APACHE) or
+[MIT](LICENSE-MIT), at your option.
+
+Some files retain their own copyright notice, however, for full authorship
```
nit: s/, however/. However,

valentinewallace commented on LICENSE.md:
```diff
@@ -0,0 +1,14 @@
+Some files retain their own copyright notice, however, for full authorship
+information, see version control history.
+
+Except as otherwise noted in individual files, all files in this repository are
+licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+http://opensource.org/licenses/MIT>, at your option.
```
Is this how you intended the formatting of the links to look? https://i.imgur.com/V3mnHRF.png

TheBlueMatt replied:
It was copied from rust at the time, I updated it to the newer version that rust has.

valentinewallace commented on LICENSE-MIT:
```diff
@@ -0,0 +1,16 @@
+Permission is hereby granted, free of charge, to any person obtaining a copy of
```
According to this: https://opensource.org/licenses/MIT isn't this license missing `Copyright <YEAR> <COPYRIGHT HOLDER>` at the top?

TheBlueMatt replied:
I don't believe its required - rust doesn't have it on theirs.

TheBlueMatt commented:
Squashed with no diff.
